tests/containers/multi_runtime.pm

```diff
@@ -0,0 +1,174 @@
+                $network = "ip6net";
+                assert_script_run "$sudo $runtime network create --ipv6 $network";
+                assert_script_run "$sudo $runtime network inspect $network";
+                $network = "--network $network";
+            }
+
+            script_retry("$sudo $runtime pull $test_image", retry => 3, delay => 60, timeout => 180);
+            assert_script_run "$sudo $runtime run -d --name $name $network -p $port:80 $volumes $test_image";
+            push @containers, {name => $name, runtime => $runtime, sudo => $sudo, port => $port};
+            $port++;
+        }
+    }
+
+    # Test connectivity to all containers on localhost
+    for my $container (@containers) {
+        next if ($container->{name} eq "test_root_podman_6");    # XXX
+        assert_script_run "curl $curl_opts http://localhost:$container->{port}",
+          fail_message => "failed IPv$ip_version localhost test for $container->{name}";
+    }
+
+    my $iface = script_output "ip -$ip_version --json route list match default | jq -r '.[0].dev'";
+    my $ip_addr = script_output "ip -$ip_version --json addr show $iface | jq -r '.[0].addr_info[0].local'";
+    $ip_addr = "[$ip_addr]" if ($ip_version == 6);
+
+    # Test connectivity to all containers on IP address
+    for my $container (@containers) {
+        assert_script_run "curl $curl_opts http://$ip_addr:$container->{port}",
+          fail_message => "failed IPv$ip_version test for $container->{name}";
+    }
+
+    # XXX This test doesn't seem to work on o3
+    if ($ip_version != 6) {
+        # Test container connectivity to the outside world
+        for my $container (@containers) {
+            # This doesn't seem to work on o3
+            assert_script_run "$container->{sudo} $container->{runtime} exec $container->{name} curl $curl_opts $test_url",
+              fail_message => "failed IPv$ip_version Internet test for $container->{name}";
+        }
+    }
+
+    # Cleanup
+    for my $container (@containers) {
+        assert_script_run "$container->{sudo} $container->{runtime} rm -vf $container->{name}";
+        assert_script_run "$container->{sudo} $container->{runtime} rmi $test_image";
+    }
+}
+
+sub run {
+    select_serial_terminal;
+
+    my @pkgs = ("docker", "podman", "jq");
+    install_packages(@pkgs);
+
+    # https://docs.docker.com/engine/daemon/ipv6/
+    assert_script_run "sed -i 's%^{%&\"ipv6\":true,\"fixed-cidr-v6\":\"2001:db8:1::/64\",%' /etc/docker/daemon.json";
+    record_info("docker daemon.json", script_output("cat /etc/docker/daemon.json"));
+    systemctl "enable --now docker";
+
+    record_info("docker root", script_output("docker info"));
+    record_info("podman root", script_output("podman info"));
+
+    # Needed to avoid:
+    # WARNING: COMMAND_FAILED: '/sbin/iptables -t nat -F DOCKER' failed: iptables: No chain/target/match by that name.
+    # See https://bugzilla.suse.com/show_bug.cgi?id=1196801
```

Comment:
Not sure if this contributes to a better test or not, but I think this deserves a softfailure. WDYT?

Reply:
I will add an unconditional soft-failure because having any subtest fail and check status on each script_run would be tricky.

Reply:
Added a soft-failure when DOCKER rules are detected in iptables.